include/mxnet/base.h

```diff
@@ -200,7 +208,7 @@ struct Context {
     return true;
   }
   /*! \brief the maximal device type */
-  static const int32_t kMaxDevType = 4;
+  static const int32_t kMaxDevType = 6;
```

Comment:
It's already here, but generally I see this sort of number derived from the last enum value, such as DevTypeMax or something.   Then you don't have to maintain it.

enum DeviceType {
      kCPU = cpu::kDevMask,
      kGPU = gpu::kDevMask,
      kCPUPinned = 3,
      kCPUShared = 5,
      kDeviceType /* or something like that */
    };

By the way, what happened to 4?

Reply:
Its a mask, so 101